### ISSUE TITLE
MSRV policy

### ISSUE THREAD

**Issue by jplatte** [mozilla/uniffi-rs#1242]
Hi! I'm curious what your thoughts are on what Rust version UniFFI should minimally support. I found that it currently contains a [hack](https://github.com/mozilla/uniffi-rs/blob/52c38d3eecf88a83b46d624066fd6f79288b1375/uniffi_bindgen/src/scaffolding/templates/ReexportUniFFIScaffolding.rs) for a [bug](https://github.com/rust-lang/rust/issues/50007) in stable Rust (include 1.61.0 released today) that should be fixed in 1.62.0 ([PR](https://github.com/rust-lang/rust/pull/95604)).

I would like to remove this hack ahead of posting a PR for a macro frontend (see recent discussion in #385 for plans around that). Is it acceptable to increase the MSRV to 1.62.0 (beta) for that now based on the assumption that there isn't going to be a crates.io release in the next 6 weeks (given the last release is only 2w old)? Or if not now, would it be okay to bump once 1.62.0 is stable?

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/UNIFFI-164)
┆friendlyId: UNIFFI-164

**Comment by mhammond** [comment]
Our policy is at https://github.com/mozilla/uniffi-rs/blob/main/docs/policies/rust-versions.md, which implies 1.59.0 is the min version we want to support currently.

**Comment by jplatte** [comment]
Given the current `rust-toolchain.toml` contains 1.60 (https://github.com/mozilla/uniffi-rs/blob/main/rust-toolchain.toml), that seems unlikely 😄

Right now it seems to me like it should take at most a few months until MSRV can be 1.62, which is probably fine for upstreaming my changes (at least from my / Element's perspective).

**Comment by mhammond** [comment]
> Given the current `rust-toolchain.toml` contains 1.60 (https://github.com/mozilla/uniffi-rs/blob/main/rust-toolchain.toml), that seems unlikely 😄

Right - that's the "uses" version, which m-c is also currently on. The m-c "requires" version is 1.59 - often there's a larger gap but it's small now. So I'm not quite sure what's "unlikely" - is there something we need to clarify in the docs?

(Regardless, given the policy exists, I'm going to close this, but happy to keep engaging on this)

**Comment by jplatte** [comment]
Yeah I saw this big table with different versions in the moz documentation and understood none of it. What's "requires" vs "uses"?

**Comment by mhammond** [comment]
I'll update the docs to make that clearer, but:
* uses is what all the CI and release engineering machines actually use.
* requires is that some downstream packagers (eg, some linux distros) use, so the min required for everything to "work"
